### PR TITLE
flac: add 1.4.2

### PR DESCRIPTION
**Pull request by chausner**
Specify library name and version:  **flac/1.4.2**

This PR continues the work started by @Croydon in https://github.com/conan-io/conan-center-index/pull/13800. The problem in that older PR was that the patches had not been applied.

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by Croydon — APPROVED** [review]
Thank you!

**Review by jwillikers — DISMISSED** [review]

File: recipes/flac/all/conanfile.py [hunk 1]
```diff
@@ -97,16 +102,17 @@ def package_info(self):
...
-        self.cpp_info.filenames["cmake_find_package"] = "flac"
-        self.cpp_info.filenames["cmake_find_package_multi"] = "flac"
-        self.cpp_info.names["cmake_find_package"] = "FLAC"
-        self.cpp_info.names["cmake_find_package_multi"] = "FLAC"
-        self.cpp_info.components["libflac"].names["cmake_find_package"] = "FLAC"
-        self.cpp_info.components["libflac"].names["cmake_find_package_multi"] = "FLAC"
-        self.cpp_info.components["libflac++"].names["cmake_find_package"] = "FLAC++"
-        self.cpp_info.components["libflac++"].names["cmake_find_package_multi"] = "FLAC++"
+        # TODO remove once conan v2 is the only support and recipes have been migrated
+        if Version(conan_version) < "2.0.0-beta":
```
Comment (prince-chrismc) [inline recipes/flac/all/conanfile.py:106]: This is not needed, the old problems where fixed in more recent betas,
Please remove this 🙏

**Review by prince-chrismc — COMMENTED** [review]
Few extras for this recipe we can sneak in 🙏 

https://github.com/conan-io/conan-center-index/pull/15372/files#diff-46e0f445c2645cf966c3341c9e8b009d71467ca87c6418b7a824fa6a9c333559R39 should use `rm_safe` (bump min conan to 1.53)

https://github.com/conan-io/conan-center-index/pull/15372/files#diff-46e0f445c2645cf966c3341c9e8b009d71467ca87c6418b7a824fa6a9c333559R30-R31
can be replace with the new `export_conandata_patches` helper

File: recipes/flac/all/conanfile.py (hunk 1 above)
Comment (chausner) [reply recipes/flac/all/conanfile.py]: Like so? https://github.com/conan-io/conan-center-index/pull/15372/commits/91c53fcd96e242b196780ecdbee24c9fc0645038

**Comment by chausner** [comment]
> Few extras for this recipe we can sneak in 🙏
> 
> https://github.com/conan-io/conan-center-index/pull/15372/files#diff-46e0f445c2645cf966c3341c9e8b009d71467ca87c6418b7a824fa6a9c333559R39 should use `rm_safe` (bump min conan to 1.53)
> 
> https://github.com/conan-io/conan-center-index/pull/15372/files#diff-46e0f445c2645cf966c3341c9e8b009d71467ca87c6418b7a824fa6a9c333559R30-R31 can be replace with the new `export_conandata_patches` helper

Addressed in 9bda74c4ee831edbe1ac29ba787da899a31f3ca0 and 955ef84abb67216777c78a1c73ba59e8370b0937

**Review by jwillikers — APPROVED** [review]

**Review by prince-chrismc — DISMISSED** [review]

**Comment by prince-chrismc** [comment]
image

**Review by Croydon — APPROVED** [review]

File: recipes/flac/all/conanfile.py [hunk 2]
```diff
@@ -1,10 +1,11 @@
 from conan import ConanFile
 from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
 from conan.tools.env import VirtualBuildEnv
-from conan.tools.files import apply_conandata_patches, copy, get, rmdir
+from conan.tools.files import apply_conandata_patches, export_conandata_patches, copy, get, rmdir
+from conan.tools.scm import Version
 import os
 
-required_conan_version = ">=1.47.0"
+required_conan_version = ">=1.53.0"
```
Comment (SpaceIm) [inline recipes/flac/all/conanfile.py:8]:
```suggestion
required_conan_version = ">=1.54.0"
```

File: recipes/flac/all/conanfile.py [hunk 3]
```diff
@@ -56,6 +56,8 @@ def generate(self):
         tc.variables["BUILD_EXAMPLES"] = False
         tc.variables["BUILD_DOCS"] = False
         tc.variables["BUILD_TESTING"] = False
+        # Honor BUILD_SHARED_LIBS from conan_toolchain (see https://github.com/conan-io/conan/issues/11840)
+        tc.cache_variables["CMAKE_POLICY_DEFAULT_CMP0077"] = "NEW"
```
Comment (SpaceIm) [inline recipes/flac/all/conanfile.py:60]:
```suggestion
```

File: recipes/flac/all/conanfile.py [hunk 4]
```diff
@@ -101,7 +105,6 @@ def package_info(self):
         self.output.info("Appending PATH environment variable: {}".format(bin_path))
         self.env_info.PATH.append(bin_path)
 
-        # TODO: to remove in conan v2 once cmake_find_package_* generators removed
         self.cpp_info.filenames["cmake_find_package"] = "flac"
```
Comment (SpaceIm) [inline recipes/flac/all/conanfile.py:108]:
```suggestion
        # TODO: to remove in conan v2
        self.cpp_info.filenames["cmake_find_package"] = "flac"
```

**Review by SpaceIm — CHANGES_REQUESTED** [review]
this PR suffered from CI issues, so here some extra reviews to take advantage that another build is required.

**Comment by chausner** [comment]
> this PR suffered from CI issues, so here some extra reviews to take advantage that another build is required.
> 
> * CMAKE_POLICY_DEFAULT_CMP0077 not required since conan 1.54.0
> * keep the TODO to track conan v1 stuff

Thank you, I applied all your suggestions.

**Review by SpaceIm — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**